src/Doctrine/Orm/State/LinksHandlerTrait.php

```diff
@@ -120,7 +120,14 @@ private function handleLinks(QueryBuilder $queryBuilder, array $identifiers, Que
             foreach ($identifierProperties as $identifierProperty) {
                 $placeholder = $queryNameGenerator->generateParameterName($identifierProperty);
                 $queryBuilder->andWhere("$joinAlias.$identifierProperty = :$placeholder");
-                $queryBuilder->setParameter($placeholder, $this->getIdentifierValue($identifiers, $hasCompositeIdentifiers ? $identifierProperty : null), $doctrineClassMetadata->getTypeOfField($identifierProperty));
+
+                if ($doctrineClassMetadata->hasAssociation($link->getToProperty())) {
```

Comment:
I think that we really don't want to go there especially here. Please check the `IdentifiersExtractor` class instead? Also can you provide a failing functional test case, I'm pretty sure I can cover that case in a jiffy.